pageserver/src/layered_repository/timeline.rs

```diff
@@ -432,14 +432,12 @@ impl LogicalSize {
                 .map(CurrentLogicalSize::Exact)
             }
             None => {
-                let non_negative_size_increment = size_increment.max(0);
-                u64::try_from(non_negative_size_increment)
-                    .with_context(|| {
-                        format!(
-                            "Failed to convert size increment {non_negative_size_increment} to u64"
-                        )
-                    })
-                    .map(CurrentLogicalSize::Approximate)
+                let non_negative_size_increment = if size_increment < 0 {
+                    0
+                } else {
+                    u64::try_from(size_increment).expect("not negative, cannot fail")
```

Comment:
Maybe worth mentioning that compiler should be able to remove panic code, but up to you, it may be redundant

Reply:
Given how outdated the comments may become after big refactorings, I'd rater keep it as is, it's pretty self-explanatory and atomic enough.